bot/exts/nominations.py

```diff
@@ -0,0 +1,81 @@
+import re
+from typing import Optional
+
+import discord
+from discord.ext import commands
+
+from bot import constants, logger
+from bot.bot import ThreadBot
+from bot.utils.helpers import get_thread_from_message_id
+
+NOMINATION_MESSAGE_REGEX = re.compile(
+    r"<@!?\d+> \((.+)#\d{4}\) for Helper!\n\n\*\*Nominated by:\*\*",
+    re.MULTILINE
+)
+
+# When nominations are posted manually, the Discord message box converts +1/-1 to the thumbs emojis
+NOMINATION_ENDING_TEXT = "react 👍 for approval, or 👎 for disapproval*."
+
+
+class Nominations(commands.Cog):
+    """Cog for creating an archiving nomination threads when votes are posted/archived."""
+
+    def __init__(self, bot: ThreadBot) -> None:
+        self.bot = bot
+        # Temporarily stores the name of the member who's vote just got posted.
+        # Need to cache this as votes can span multiple messages (due to long nomination reasons).
+        self.nominated_member_name: Optional[str] = None
+
+        if constants.DEBUG_MODE:
+            self.archive_time = constants.ThreadArchiveTimes.DAY.value
+        else:
+            self.archive_time = constants.ThreadArchiveTimes.WEEK.value
+
+    @commands.Cog.listener()
+    async def on_message(self, message: discord.Message) -> None:
+        """Create a thread on votes sent in the nominations voting channel."""
+        if message.channel.id != constants.Channels.nomination_voting:
+            return  # Ignore messages not in the voting channel
+
+        if match := NOMINATION_MESSAGE_REGEX.match(message.content):
+            self.nominated_member_name = match.group(1)
+
+        if not message.content.endswith(NOMINATION_ENDING_TEXT):
+            return  # Only create thread on final message, as a vote could be split into multiple messages
```

Comment:
This should fit the line length?
```suggestion
            return  # Votes could be split into multiple messages, create the thread on the final message
```